.github/actions/build/action.yml

```diff
@@ -0,0 +1,120 @@
+name: build
+description: 'build connaisseur image'
+inputs:
+  image-repo:
+    description: 'image repository consisting of registry and repository'
+    required: true
+  image-tag:
+    description: 'image tag containing the tag to be used for the image-repo'
+    required: true
+  repo-owner:
+    description: 'name of repository owner "github.repository_owner"'
+    required: true
+  github-token:
+    description: 'github token via "secrets.GITHUB_TOKEN"'
+    required: true
+  cosign-version:
+    description: 'cosign version to be used'
+    required: true
+  cosign-private-key:
+    description: 'cosign private key'
+    required: true
+  cosign-password:
+    description: 'cosign private key password'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install Cosign
+      uses: sigstore/cosign-installer@main
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Login with ghcr
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ inputs.repo-owner }}
+        password: ${{ inputs.github-token }}
+    - name: Build and Push
+      id: build
+      uses: docker/build-push-action@v4
+      with:
+        push: true
+        tags: ${{ inputs.image-repo }}:${{ inputs.image-tag }}
+        file: docker/Dockerfile
+        build-args: COSIGN_VERSION=${{ inputs.cosign-version }}
+        sbom: true
+        provenance: true
+    - name: Sign image with a key
```

Comment:
We did in the past conciously decide against signing the image with the notary key within the pipeline. Reasons were that we run stuff within the pipeline that's not under our control and accept PRs from contributors who could then exfil the key, making the signatures void. Those arguments still seem to apply

Reply:
Yes, I agree. However, this does not replace the release signing so far, but only signs for the internal pipeline for now. This may evolve into a pipeline that allows for automated signed releases and then will have to either reconsider or require an additional signature by an offline key.
However, so far this only affects the intermediate testing images and we could for example consider to make the nightly images public with a "less secure" key that is stored in github. Still, the images uploaded so far are private and not publicly available